src/function/scalar/sequence/nextval.cpp

```diff
@@ -122,7 +122,8 @@ static unique_ptr<FunctionData> NextValBind(ClientContext &context, ScalarFuncti
 		auto seqname = ExpressionExecutor::EvaluateScalar(*arguments[0]);
 		if (!seqname.IsNull()) {
 			auto qname = QualifiedName::Parse(seqname.ToString());
-			sequence = Catalog::GetCatalog(context).GetEntry<SequenceCatalogEntry>(context, qname.schema, qname.name);
+			sequence = Catalog::GetCatalog(context).GetEntry<SequenceCatalogEntry>(context, qname.schema, qname.name,
+			                                                                       true /* if exists */);
```

Comment:
there was an ongoing discussion on a previous implementation that this would cause an additional catalog lookup. there might however be a bug in the current duckdb that does not cache the nextval, because I checked through breakpoints and there's 2 lookups for each INSERT anyway, regardless if we do sequence + creation or serial. so I don't think this additional parameter is the issue here... I am in process of actually proving this lol, but I am still trying to figure out a smart way to benchmark it

Reply:
Perhaps do a bulk insert from e.g. a `generate_series`, both with and without the default value? I imagine the catalog look-up on every tuple would slow down the default value computation significantly. You can look at e.g. `benchmark/tpch/csv/read_lineitem_csv.benchmark` for an example benchmark that times appends.